src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs

```diff
@@ -759,8 +764,10 @@ public Task<RunspaceHandle> GetRunspaceHandleAsync(CancellationToken cancellatio
                         return shell.Invoke<TResult>(null, invocationSettings);
                     }
 
-                    // May need a cancellation token here
-                    return await Task.Run<IEnumerable<TResult>>(() => shell.Invoke<TResult>(input: null, invocationSettings), CancellationToken.None).ConfigureAwait(false);
+                    // This is the primary reason that ExecuteCommandAsync takes a CancellationToken
+                    return await Task.Run<IEnumerable<TResult>>(
+                        () => shell.Invoke<TResult>(input: null, invocationSettings), cancellationToken)
+                        .ConfigureAwait(false);
```

Comment:
Looking at this makes me think we should also contemplate linking the cancellation token to the `shell.Stop()` call

Reply:
That would be a good idea...how would we go about doing that?

Reply:
Something like this:

```csharp
cancellationToken.Register(() => shell.Stop());

return await Task.Run<IEnumerable<TResult>>(input: null, invocationSettings), cancellationToken).ConfigureAwait(false);
```